src/ui/components/Library/Team/View/Tests/hooks/useTests.ts

```diff
@@ -24,3 +30,92 @@ export function useTests(): Test[] {
   );
   return value;
 }
+
+export function useTest(testId: string) {
+  const { teamId } = useContext(TeamContext);
+  const { data, loading } = useQuery(GET_TEST, {
+    variables: { workspaceId: teamId, testId },
+  });
+
+  if (loading) {
+    return { test: null, loading: true };
+  }
+  const test =
+    {
+      ...data.node.tests.edges[0].node,
+      executions: data.node.tests.edges[0].node.executions.map(
+        (e: GetTestForWorkspace_node_Workspace_tests_edges_node_executions) => ({
+          ...e,
+          recording: e.recording.map(
+            (
+              r: GetTestForWorkspace_node_Workspace_tests_edges_node_executions_recording | null
+            ) => {
+              if (!r) {
+                return r;
+              }
+
+              return { ...r, id: r.uuid };
+            }
+          ),
+        })
+      ),
+    } ?? {};
+
+  const processedTest = {
+    ...test,
+    failureRate: getFailureRate(test.executions),
+    failureRates: getFailureRates(test.executions),
+    errorFrequency: getErrorFrequency(test.executions),
+  };
+
+  return { test: processedTest, loading };
+}
+
+function getFailureRates(executions: TestExecution[]) {
+  return {
+    hour: getFailureRate(
+      executions.filter(e => (Date.now() - new Date(e.createdAt).getTime()) / 1000 / 60 / 60 < 1)
+    ),
+    day: getFailureRate(
+      executions.filter(
+        e => (Date.now() - new Date(e.createdAt).getTime()) / 1000 / 60 / 60 / 24 < 1
+      )
+    ),
+    week: getFailureRate(
+      executions.filter(
+        e => (Date.now() - new Date(e.createdAt).getTime()) / 1000 / 60 / 60 / 24 < 7
+      )
+    ),
+    month: getFailureRate(
+      executions.filter(
+        e => (Date.now() - new Date(e.createdAt).getTime()) / 1000 / 60 / 60 / 24 < 30
+      )
+    ),
+  };
+}
+
+function getErrorFrequency(executions: TestExecution[]) {
+  return executions.reduce((acc: Record<string, number>, e) => {
+    let newAcc = { ...acc };
```

Comment:
:hand: no need to spread this. It starts with a brand new empty object at the bottom, so it's safe to mutate all the way through the `reduce()` call.